app/importers/activity_insight_importer.rb

```diff
@@ -166,7 +166,12 @@ def call
           if pi.persisted?
             update_pub_record(pub_record, pub)
           else
-            pi.save!
+            unless pub_record.doi_verified == true
+              pub_record.oa_workflow_state = 'automatic DOI verification pending'
+              pub_record.save!
+              DOIVerificationJob.perform_later(pub_record.id)
+              pi.save!
+            end
```

Comment:
This is also going to save a workflow state for things that don't belong in the workflow. If you want it to be here, you could put a conditional around setting the state here. Alternate could be to leave that logic where it was & have `unless pub_record.doi_verified == true` be `unless pub_record.doi_verified == true || pub_record.oa_workflow_state == 'automatic DOI verification pending' to avoid queueing the job multiple times

Reply:
I like the change of using `#update` here. I think `#can_receive_new_ai_oa_files?` covers most of what we want, but it doesn't account for whether there is an activity insight file location. Easy fix could be to just move `activity_insight_file_location = pub.postprints&.first&.location` up & add `activity_insight_file_location.present?` to the current conditional. We have a a publication scope `activity_insight_oa_publication` that would cover all conditions, but I'm not sure if that would be useful since we would want to check `pub` for a file location and `pub_record` for conditions covered by `#can_receive_new_ai_oa_files?`